xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java

```diff
@@ -104,7 +109,7 @@ public float getLeafValue(FVec feat, int root_id) {
         }
     }
 
-    static class Node implements Serializable {
+    public static class Node extends RegTreeNode implements Serializable {
```

Comment:
looks weird at first sight: basic extends specific?

Reply:
Oh yeah. Wanted to keep the original naming. The full name is RegTreeImpl.Node which is not that specific. But you are **right**.

Waiting for Michal, but yes I'll try to rename it if it won't break anything.

Reply:
please do not rename Node, rename RegTreeNode instead if you need to

Reply:
As the fully qualified name of "Node" is not node, but RegTreeImpl.Node (without package of course) I'd leave it as-is.